submitit/core/submission.py

```diff
@@ -53,9 +53,12 @@ def process_job(folder: Union[Path, str]) -> None:
         env = job_environment.JobEnvironment()
         env._handle_signals(paths, delayed)
         result = delayed.result()
+        logger.info("Job completed successfully")
+        del delayed  # if it blocks here, you have a race condition that must be solved!
```

Comment:
I'm deleting explicitely before dumping the results to make sure any hang there would be visible

Reply:
Why would it block ? what issues are you trying to solve ?

Reply:
ok so this is about some jobs that deadlock themselves and don't exit normally. This is a user error, but since submitit was properly writing the result file, they would never notice it, and would left zombie jobs running behind.